handling-null-values-in-avro-and-protobuf/kafka/README.md

```diff
@@ -0,0 +1,195 @@
+        }
+```
+
+In the `AvroConsumer.java` file, we'll consume those events and print them to the console:
+
+```java
+avroConsumer.subscribe(Collections.singletonList("avro-purchase"));
+
+            ConsumerRecords<String, PurchaseAvro> avroConsumerRecords = avroConsumer.poll(Duration.ofSeconds(2));
+            avroConsumerRecords.forEach(avroConsumerRecord -> {
+                PurchaseAvro avroPurchase = avroConsumerRecord.value();
+                System.out.print("Purchase details consumed from topic with Avro schema { ");
+                System.out.printf("Customer: %s, ", avroPurchase.getCustomerId());
+                System.out.printf("Total Cost: %f, ", avroPurchase.getTotalCost());
+                System.out.printf("Item: %s } %n", avroPurchase.getItem());
+
+            });
+
+```
+
+Copy and paste the contents of `path to file/s on GitHub` into the `AvroConsumer.java` and `AvroProducer.java` files.
+
+Create a file at `handling-null-values/kafka/code/src/main/avro/purchase.avsc`. 
+
+Copy and paste the following into that file:
+
+```
+{
+  "type":"record",
+  "namespace": "io.confluent.developer.avro",
+  "name":"PurchaseAvro",
+  "fields": [
+    {"name": "item", "type": "string" },
+    {"name": "total_cost", "type": "double" },
+    {"name": "customer_id", "type": "string"}
+  ]
+}
+```
+
+If you run the code using `./gradlew runAvroProducer`, you will see that the producer "hangs" and does not produce events.
+
+For the "item" field in `purchase.avsc`, put:
+
+```
+{
+  "type":"record",
+  "namespace": "io.confluent.developer.avro",
+  "name":"PurchaseAvro",
+  "fields": [
+    {"name": "item", "type": ["string", "null"] },
+    {"name": "total_cost", "type": "double" },
+    {"name": "customer_id", "type": "string"}
+  ]
+}
+```
+When you run `./gradlew runAvroProducer` and furthermore, `./gradlew runAvroConsumer`, you'll see that the events with null items are produced and consumed successfully. 
+
+Next, copy and paste the contents of `path to file/s on GitHub` into the `ProtoConsumer.java` and `ProtoProducer.java` files.
+
+Create a file at `handling-null-values/kafka/code/src/main/proto/purchase.proto`. 
+
+Copy and paste the following into that file:
+
+```
+syntax = "proto3";
+
+package io.confluent.developer.proto;
+option java_outer_classname = "PurchaseProto";
+
+message Purchase {
+  string item = 1;
+  double total_cost = 2;
+  string customer_id = 3;
+}
+```
+
+Look at `ProtoProducerApp.java`, lines 76-77:
+
+```java
+        purchaseBuilder.setCustomerId("Customer Null")
+                .setTotalCost(random.nextDouble() * random.nextInt(100));
+``` 
+
+We can see that the developer who wrote this app 'forgot' to write the `setItem()` method that adds an item. This means that the value will be null. But when you run you run `./gradlew runProtoProducer` and `./gradlew runProtoConsumer` no errors will arise. That's because Protobuf automatically handles default values. 
```

Comment:
what makes it into Kafka in this case? empty string? suggest saying that here maybe with a link to protobuf doc about default values per type